### PR TITLE
🔥py38 + torch 2 🔥🔥🔥🚀

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Title is all we need.

There is one line in `setup.py` I don't know if I need to change and if yes how.

```python3
deps["importlib_metadata"] + ";python_version<'3.8'",  # importlib_metadata for Python versions that don't have it
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
     # Ensure running with CircleCI/huggingface
     check_circleci_user:
         docker:
-            - image: cimg/python:3.7.12
+            - image: cimg/python:3.8.12
```
Comment (ydshieh) [inline .circleci/config.yml:15]: 3.7.12 -> 3.8.x  ==> `x = 12`, right?

**Comment by ydshieh** [comment]
The whole suite of tests is run and passed in this [run](https://app.circleci.com/pipelines/github/huggingface/transformers/60009/workflows/73d754f8-017d-458f-8dc2-c6166d30e1de)

**Review by sgugger — APPROVED** [review]
Thanks!

File: tests/trainer/test_trainer.py [hunk 2]
```diff
@@ -1855,6 +1855,7 @@ def test_torchdynamo_full_eval(self):
         self.assertAlmostEqual(metrics["eval_loss"], original_eval_loss)
         torchdynamo.reset()
 
+    @unittest.skip("torch 2.0.0 gives `ModuleNotFoundError: No module named 'torchdynamo'`.")
```
Comment (amyeroberts) [inline tests/trainer/test_trainer.py:1858]: Does this mean `require_torchdynamo` doesn't work for torch 2.0.0?

File: tests/onnx/test_onnx_v2.py [hunk 3]
```diff
@@ -499,6 +500,7 @@ def test_tensorflow_export_seq2seq_with_past(
 class StableDropoutTestCase(TestCase):
     """Tests export of StableDropout module."""
 
+    @unittest.skip("torch 2.0.0 gives `torch.onnx.errors.OnnxExporterError: Module onnx is not installed!`.")
```
Comment (amyeroberts) [inline tests/onnx/test_onnx_v2.py:503]: I realise this is more of a torch issue - but perhaps we want to open an issue to keep track and update this?

**Review by amyeroberts — APPROVED** [review]
Thanks! 🚀

File: tests/trainer/test_trainer.py (hunk 2 above)
Comment (ydshieh) [reply tests/trainer/test_trainer.py:1858]: But in this PR, I would just focus on make what needs to be done for CI to run with torch 2. We can do housekeeping later :-)

**Pull request merged**